_docs/contributing_to_a_module.md

```diff
@@ -0,0 +1,295 @@
+
+# Testing
+## Getting Started
+
+Our Puppet modules provide a Gemfile in the module root directory, we can then use the pdk to install all of the Gems listed in the Gemfile.
+
+which can tell a Ruby package manager such as [bundler](http://bundler.io/) what Ruby packages,
+or Gems, are required to build, develop, and test this software.
+
+Please make sure you have the [pdk](https://puppet.com/try-puppet/puppet-development-kit/) on your system, and then use it to
+install all dependencies needed for this project in the project root by running
+
+```shell
+% pdk bundle install
+Fetching gem metadata from https://rubygems.org/........
+Fetching gem metadata from https://rubygems.org/..
+Using rake (10.1.0)
+Using builder (3.2.2)
+-- 8><-- many more --><8 --
+Using rspec-system-puppet (2.2.0)
+Using serverspec (0.6.3)
+Using rspec-system-serverspec (1.0.0)
+Using bundler (1.3.5)
+Your bundle is complete!
+Use `pdk bundle show [gemname]` to see where a bundled gem is installed.
+```
+
+NOTE: some systems may require you to run this command with sudo.
+
+If you already have those gems installed, make sure they are up-to-date:
+
+```shell
+% pdk bundle update
+```
+
+## Checking your PR automated test results
+
+When you create a PR, our automated test suite will automatically pick up your changes and run tests against them.
+This will highlight and regressions that you change may have introduced.
+
+### How do I know the status of the automated tests?
+
+At the bottom of your PR a breakdown of the automated tests running will be shown:
+
+![automated_test_suite](test_run_info.png)
+
+- A green tick (Tests have passed)
+- A red cross (Tests have failed)
+- A yellow dot (Tests are still running)
+- A grey dot (The job has been cancelled)
+
+A PR will not be merged with failing tests, if you click on the small red cross icon it will give you additional information
+on what has actually failed. Once this is addressed you can commit your fix and continue working. If your fix is difficult and
+you need to do additional debugging check out the section in this document called **Running tests on your local machine** as this
+will allow you to run interactive debugging sessions using pry.
+
+## Running Tests on your local machine
+
+With all dependencies in place and up-to-date, run the tests:
+
+### Unit Tests
+
+```shell
+% pdk bundle exec rake spec
+```
+
+This executes all the [rspec tests](http://rspec-puppet.com/) in the directories defined [here](https://github.com/puppetlabs/puppetlabs_spec_helper/blob/699d9fbca1d2489bff1736bb254bb7b7edb32c74/lib/puppetlabs_spec_helper/rake_tasks.rb#L17) and so on.
+rspec tests may have the same kind of dependencies as the module they are testing. Although the module defines these dependencies in its [metadata.json](./metadata.json),
+rspec tests define them in [.fixtures.yml](./fixtures.yml). We use this over `pdk test unit` as it provides live feedback as the tests are running. See below: 
```

Comment:
Just curious about the `metadata.json` and `.fixtures.yml` links and what they're supposed to link to?

Reply:
Good shout, initially this was per repo but I have moved it out. I will work on updating this.